.github/workflows/integration-tests-reusable.yml

```
@@ -120,8 +132,8 @@ jobs:
       env:
         XUNIT_SHARD: ${{ matrix.shard }}
         XUNIT_SHARD_COUNT: ${{ inputs.num_shards }}
-        DAFNY_RELEASE: ${{ github.workspace }}/unzippedRelease/dafny
       run: |
+        echo ${{ inputs.all_platforms }} | grep true && export DAFNY_RELEASE="${{ github.workspace }}/unzippedRelease/dafny"
```

Something like this maybe?

```suggestion
        [ "${{ inputs.all_platforms }}" == "true" ] && export DAFNY_RELEASE="${{ github.workspace }}/unzippedRelease/dafny"
```